dashboard/app/controllers/maker_controller.rb

```diff
@@ -122,17 +122,24 @@ def login_code
   # renders a page for users to copy and paste a login key
   def display_code
     # Generate encrypted code to display to user
-    user_auth = current_user.authentication_options.find_by_credential_type(AuthenticationOption::GOOGLE)
-    @secret_code = Encryption.encrypt_string_utf8(
-      Time.now.strftime('%Y%m%dT%H%M%S%z') + user_auth['authentication_id'] + user_auth['credential_type']
-    )
+    user_auth = current_user&.authentication_options&.find_by_credential_type(AuthenticationOption::GOOGLE)
+    if user_auth
+      @secret_code = Encryption.encrypt_string_utf8(
+        Time.now.strftime('%Y%m%dT%H%M%S%z') + user_auth.authentication_id + user_auth.credential_type
+      )
+    end
   end
 
   # GET /maker/confirm_login
+  # @param [String] params[:user_return_to] Where to redirect the user after they've logged in
   # Renders a page to confirm uses want to login via Google OAuth
   # This route is need to convert the GET request from the Maker App into
   # a POST that can be used to login via Google OAuth
   def confirm_login
+    session[:user_return_to] = params[:user_return_to]
```

Comment:
Now that we are handling the redirect here, I think we can remove this if statement (https://github.com/code-dot-org/code-dot-org/blob/c47eb84dd015b27155cd45b53f870daab8101029/dashboard/app/controllers/sessions_controller.rb#L10) from the sessions controller. If I recall correctly, it was added so that we could piggy-back on the session management from that route. So now that we handle it here, I think that path is unused and can be removed.

Reply:
I don't think we can do that until after we publish the new maker app right? Since the current version of maker app still goes through there?

Reply:
Yup! Thanks for catching that - I didn't think that through. Is there a way we can track that? It's a small change, but anytime we can minimize the number of branches of code we support, I'm excited.

Reply:
yeah, jillian and i were discussing this before break -- we could log to either Honeybadger (easily viewable but can cause alarm if it's triggered a lot) or Firehose (we aren't notified and have to remember to check Redshift) when this path is hit. then, if we don't log any usage for ~1 month (or whatever timeframe), we can safely remove that code